decimal64.go

```diff
@@ -102,21 +102,23 @@ func renormalize(exp int, significand uint64) (int, uint64) {
 }
 
 func rescale(exp int, significand uint64, targetExp int) (uint64, int) {
-	exp -= targetExp
-	var divisor uint64 = 1
-	for ; exp < -7 && divisor < significand; exp += 8 {
-		divisor *= 100000000
+	exp = targetExp - exp
+	mag := size(significand)
+	if exp > mag {
+		return 0, targetExp
 	}
-	for ; exp < -3 && divisor < significand; exp += 4 {
-		divisor *= 10000
-	}
-	for ; exp < -1 && divisor < significand; exp += 2 {
-		divisor *= 100
-	}
-	for ; exp < 0 && divisor < significand; exp++ {
```

Comment:
So instead of iterating though exp, which isn't bounded by the significand, we can return zero early if the exp is larger than the magnitude of the exponent, 
Then the powers of 10 global variable can be used with confidence that there isn't going to be any indexing errors